lib/utils.js

```diff
@@ -19,6 +19,35 @@ function shallowCopy(src, dest) {
   return dest;
 }
 
+function take(source, fromRight) {
+  return fromRight ? source.pop() : source.shift();
+}
+
+/**
+ * Takes a list of arguments and maps them to a list of names
+ * in an alternating either a left-to-right, or right-to-left direction.
+ */
+function resolveArgs(argumentsObject, nameMap, rightToLeft) {
+  var result = {};
+
+  var takeFromRight = rightToLeft;
+  var args = Array.prototype.slice.call(argumentsObject);
+
+  while (nameMap.length) {
+    var value = null;
+    var name = take(nameMap, takeFromRight);
+
+    if (args.length > 0) {
+      value = take(args, takeFromRight);
+      takeFromRight = !takeFromRight;
```

Comment:
I'm skeptical that this is going to work for all situations.  I'd want to look at the test coverage for every public function that uses this.  I just linked the current master of `express-stormpath` to this branch of `stormpath`, and I get a handful of failures that I don't see with the latest stable release of `stormpath`.  Can you try as well?